addons/stock/controllers/main.py

```diff
@@ -11,3 +15,32 @@ def a(self, debug=False, **k):
             return http.local_redirect('/web/login?redirect=/stock/barcode/')
 
         return request.render('stock.barcode_index')
+
+
+class StockReportController(http.Controller):
+
+    @http.route('/stock/<string:output_format>/<string:report_name>/<int:report_id>', type='http', auth='user')
```

Comment:
this is commit from 46fc922448bd8ab1602055f2a754ea8e610b5866 and we don't have time to clean it with this pr, please ignore diff from this commit :)